app/src/main/shared/services/AbstractLocalService.ts

```diff
@@ -87,28 +97,54 @@ export abstract class LocalService {
 
     public get(url: string){
         console.log('get', url);
-        return this.doFetch(this.makeUrl(url), this.makeRequestOptions('GET'))
-            .then(this.processResponse)
-            .catch(this.notifyOnErrors);
+        return this.getData(url, "GET");
     }
 
     public post(url: string, params?:any){
         console.log('post', url, params);
-        return this.doFetch(this.makeUrl(url), this.makeRequestOptions('POST', params))
+        return this.getData(url, "POST", params);
+    }
+
+    protected getCache(url: string) {
+        return this.cacheEngine.get(["localService", this.constructor.name, this.options.cache, encodeURIComponent(url)].join('.'));
+    }
+
+    protected setCache(url: string, data: any) {
+        this.cacheEngine.set(["localService", this.constructor.name, this.options.cache, encodeURIComponent(url)].join('.'), data);
```

Comment:
Lets pull out the logic to generate cache keys in the required format into it's own function, so it can be reused when getting and setting cache values.